### PR TITLE
Remove checks for various isnan functions.

### PR DESCRIPTION
**Pull request by drwells**
This PR removes

DEAL_II_HAVE_ISNAN
DEAL_II_HAVE_STD_ISNAN
DEAL_II_HAVE_UNDERSCORE_ISNAN

in favor of using std::isnan, which is a C++11 function, instead.

### CODE REVIEW

**Comment by drwells** [comment]
In reference to #4704.

**Comment by bangerth** [comment]
OK to merge once the tester is happy.

File: include/deal.II/base/config.h.in [hunk 1]
```diff
@@ -108,9 +108,6 @@
  */
 
 #cmakedefine DEAL_II_HAVE_CXX11_IS_TRIVIALLY_COPYABLE
-#cmakedefine DEAL_II_HAVE_ISNAN
-#cmakedefine DEAL_II_HAVE_STD_ISNAN
-#cmakedefine DEAL_II_HAVE_UNDERSCORE_ISNAN
```
Comment (bangerth) [inline include/deal.II/base/config.h.in]: where were they set? is there cmake magic to determine their values?

**Comment by drwells** [comment]
In my haste to submit this I forgot to include the relevant changes to the CMake configuration. They are now here.

**Comment by bangerth** [comment]
Yes, looks good now. Let's still see what the tester says.

File: include/deal.II/base/config.h.in (hunk 1 above)
Comment (tamiko) [inline include/deal.II/base/config.h.in]: @bangerth Yes, cmake expands the `#cmakedefine` statement with the help of cmake variables (in this case these three variables were set in one of our `check_*.cmake` files.)

**Pull request merged**